### PR TITLE
Add new option update_secrets to allow lazy or strict updating

### PR DESCRIPTION
**Pull request by AlanCoding**
This is my attempt to modify the solution for `tower_user` updating passwords, so that it can be rolled out generally to other modules, and so that it follows a universal API principal.

### CODE REVIEW

File: awx_collection/plugins/module_utils/tower_api.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ class TowerAPIModule(TowerModule):
         'tower': 'Red Hat Ansible Tower',
     }
     session = None
-    cookie_jar = CookieJar()
```
Comment (AlanCoding) [inline awx_collection/plugins/module_utils/tower_api.py:25]: @john-westcott-iv this particular thing wasn't getting used.

File: awx_collection/plugins/modules/tower_user.py [hunk 2]
```diff
@@ -55,6 +55,12 @@
       description:
         - Write-only field used to change the password.
       type: str
+    update_secrets:
+      description:
+        - C(true) will always password if user specifies password and API gives $encrypted$ for password.
```
Comment (beeankha) [inline awx_collection/plugins/modules/tower_user.py]: Sentence here is incomplete, looks like it should be "... will always **change** password if user specifies..."

**Review by beeankha — APPROVED** [review]
Other than minor comment re: parameter description text, this looks good to me!

**Comment by AlanCoding** [comment]
regate

**Comment by AlanCoding** [comment]
regate

**Comment by AlanCoding** [comment]
regate

**Comment by AlanCoding** [comment]
regate

**Comment by AlanCoding** [comment]
regate

**Pull request merged**